### DISCUSSION TITLE
Can't enable Social Logins for a Cross Platform App

### DISCUSSION THREAD

**Discussion by henrilp** [supabase/supabase#13093] [Questions]
In Supabase, I just realized that for an "Auth Provider" (let's say Google), I can only put one client Id.
My problem is that I am doing a cross platform Expo App, and Google needs one OAuth client PER APP.
I have to choose between my Web, iOS or Android apps, which one will be able to use Google Login, and the others won't be able to do it.

That's a feels bad bug that I spent several days on. Did someone encounters the same problem / found a workaround?

Thank you in advance. Otherwise I'm amazed by Supabase.

**Comment by markszente** [comment]
It's not a bug, per se, it's just not (yet) implemented properly. You have to select the web option, and have a common scheme for your apps across all platforms (eg. `xyz://`), and set the post-auth redirect to something like `xyz://auth/callback`. Of course, by not using the native oauth features, you app will not be the best UX-wise, but at least social logins will work. 

Edit: [as confirmed](https://github.com/orgs/supabase/discussions/13218#discussioncomment-5435475) by the staff, they're working on it.

**Comment by henrilp** [comment]
Thank you for your email. This solves the problem.

Have a great day!

Henri

**Comment by aaronksaunders** [comment]
@markszente this approach does not work for ionic apps or Cordova based apps because if you don’t set the redirect url properly it just doesn’t work. In fact if you use a web clientid google will error out… so there is no workaround at all
